crates/gpui/src/platform/linux/wayland/window.rs

```diff
@@ -417,6 +417,66 @@ impl WaylandWindowStatePtr {
         }
     }
 
+    pub fn handle_ime_commit(&self, text: String) {
+        if let Some(ref mut fun) = self.callbacks.borrow_mut().input {
+            if !fun(PlatformInput::KeyUp(crate::KeyUpEvent {
+                keystroke: crate::Keystroke::default(),
+            }))
+            .propagate
+            {
+                return;
+            }
+        }
```

Comment:
Also this part was kind of a hack to prevent panics. I am seeking another workaround for this.